frontend/lib/common-steps/ask-name.tsx

```diff
@@ -39,6 +40,26 @@ export const AskNameStep: React.FC<MiddleProgressStepProps> = (props) => {
           </>
         )}
       </SessionUpdatingFormSubmitter>
+      <SessionUpdatingFormSubmitter
+        mutation={NorentPreferredNameMutation}
+        initialState={(s) => ({
+          preferredFirstName:
+            s.norentScaffolding?.preferredFirstName ||
+            s.preferredFirstName ||
+            "",
+        })}
+        onSuccessRedirect={props.nextStep}
+      >
+        {(ctx) => (
+          <>
+            <TextualFormField
+              {...ctx.fieldPropsFor("preferredFirstName")}
+              label={li18n._(t`Preferred first name`)}
+            />
+            <ProgressButtons isLoading={ctx.isLoading} back={props.prevStep} />
+          </>
+        )}
+      </SessionUpdatingFormSubmitter>
```

Comment:
I think we discussed this on a video call at some point, but because this is a second `<SessionUpdatingFormSubmitter>`, it effectively means that it needs to be submitted separately from the "legal first name" form... but these steps are constructed in such a way that they expect only _one_ submittable form per step, so once the user fills out just one of these forms, they will immediately be moved to the next step without having a chance to fill out the other form.

I recommend that instead of making a separate form, you either make a `NorentFullAndPreferredNameMutation` that supersedes `NorentFullNameMutation` (deprecating the latter), or you just put the preferred name on an actual separate step (i.e. page).